### PR TITLE
Stream fixes

### LINKED ISSUE #32: Test infinite hanging
Some tests require `app.stop();` to be run after the `REQUIRE()` (or `ASSERT_XYZ` with the old system).

It seems that the old system allowed the rest of the code to be run if an assertion fails, allowing `app.stop();` to run, ending the test. This doesn't seem to be the case with Catch, which causes the test to basically run in an infinite loop if a test that requires `app.stop();` fails.

### PR DESCRIPTION
**Pull request by The-EDev**
This PR adds a specific method to stream a string (rather than an input stream) and makes it so that the test for body streaming actually tests the streamed data (or at least a part of it).

This should reduce memory usage from `body*2` at all times (when the body is streamed) to `body` at the start, decreasing to `0` by the end

oh it also changes all `REQUIRE` to `CHECK` which fixes #32 

Related to #21

### CODE REVIEW

**Comment by The-EDev** [comment]
it seems the whole thing with streaming data makes it a lot more difficult to read that data reliably (I've been struggling with this for about 12 or so hours), so for now the proper tests will have to wait (I'll add a new commit in a few hours)

File: include/crow/http_response.h [hunk 1]
```diff
@@ -209,21 +208,54 @@ namespace crow
...
-                            return false;
-                        }
-                        else
-                        {
-                            CROW_LOG_ERROR << ec << " - happened while sending body";
-                            this->end();
-                            return true;
-                        }
-                    });
+                        buf = std::string();
```
Comment (mrozigor) [inline include/crow/http_response.h:233]: Why set it to new object and then assign `is`?

Comment (The-EDev) [reply include/crow/http_response.h:233]: it seems one of my debugging lines made it here, my bad

**Comment by The-EDev** [comment]
ok looking at all test results, it's very clear the data is not being received properly, I'll probably need to find a way to synchronize the sending or at least have the receiving end wait

**Comment by The-EDev** [comment]
if this commit fails I'm gonna need to find a way to debug the test

**Comment by The-EDev** [comment]
ATM I'm looking into replacing the current implementation with one that uses `asio::streambuf` and `transfer_all()`, because multiple hours looking into this whole thing, the test is failing because 1 or 2 characters are missing from the end result.

**Review by mrozigor — APPROVED** [review]

**Pull request merged**